### DISCUSSION TITLE
Allow to use it with local domains

### DISCUSSION THREAD

**Discussion by ghost** [nextcloud/all-in-one#842] [Ideas] [answered]
### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments. 

### Steps to reproduce
1. Follow the install guide at https://github.com/nextcloud/all-in-one or https://github.com/nextcloud/all-in-one/blob/main/reverse-proxy.md
2. Go to Nextcloud AIO config at port 8080
3. Type the (local) domain name that will be used into "Please type in the domain that will be used for Nextcloud if you want to create a new instance:" field.
4. Wait for error "The server is not reachable on Port 443. You can verify this e.g. with 'https://portchecker.co/' by entering your domain there as ip-address and port 443 as port."

### Expected behavior 
It should be possible to use Nextcloud AIO local only without having it open from the outside. 

### Actual behavior 
Because server is not public, it is not possible to continue setup

### Host OS 
Ubuntu Server 20.04.4 LTS

#### Nextcloud AIO version 
Nextcloud AIO v1.4.1

#### Current channel 
latest

**Comment by szaimen** [comment]
This is a feature request as local domains are not supported

**Comment by phatpaul** [comment]
Is there just a way to bypass the domain setup part?  I like the idea of this, but I already have the reverse proxy and SSL certs handled.  I also just want to play with it locally before exposing it to the outside.

**Reply by szaimen** [reply]
The problem is that this feature is simply not implemented as everything is running on TLS by default currently.

However you may have a look at this: https://github.com/nextcloud/all-in-one/tree/main/manual-install

Using a different port than 443 as APACHE_PORT will disable TLS automatically...

**Reply by gclawes** [reply]
It's perfectly possible to set up TLS on internal domains.  It's common practice for internal corporate networks, and even possible with tools like [smallstep certificates](https://github.com/smallstep/certificates) to host a private CA that clients can trust.

**Reply by gclawes** [reply]
If the `all-in-one` container needs to trust a custom CA, I could see 3 options:
1. Mount system root trust into the AIO container (assuming the system is debian-based, or system root trust is compatible with the one on the container)
2. Add the ability in AIO to disable TLS validation.
3. Add the ability to upload a custom CA cert to AIO.

**Reply by szaimen** [reply]
Seems like someone managed to deploy it locally already before the private ip-address check was introduced. See https://github.com/nextcloud/all-in-one/issues/745#issuecomment-1167218452

The private ip-address check will be removed again when running behind reverse proxies with https://github.com/nextcloud/all-in-one/pull/859

**Reply by dantefromhell** [reply]
Caddy supports auto-generation of TLS for "local domains" (would not cover the internal corporate network scenario). See chapter "Local HTTPS" https://caddyserver.com/docs/automatic-https
That could/ should help w/ the OPs scenario.

Self-signed certs can create problems w/ verification though, depending on where the client is located. If all TLS clients are in Browsers one can override the TLS errors manually. In case any of the containers initiates TLS requests to any other container things get complicated, as you either need to

1. include the custom root CA in every container that initiates TLS requests
2. Disable TLS verification (bad idea)

@szaimen 
I am not sure I understand all bit & pieces of nextcloud-aio correctly, but there are situations in which one container attempts to use TLS to connect to another container, correct?

**Reply by szaimen** [reply]
I would prefer a solution like this: https://github.com/nextcloud/all-in-one/issues/745#issuecomment-1167218452
(it only needs to be documented correctly)

**Reply by dantefromhell** [reply]
I dont understand how does my question about TLS in between containers (and local domains) relates to internal IP addresses. Can you please elaborate?

**Comment by szaimen** [comment]
I am adding an option to disable the domain validation with https://github.com/nextcloud/all-in-one/pull/873. Not saying that it will automatically work afterwards with local domains but you may be a step closer figuring out what needs to be done. PRs are welcome!

**Comment by szaimen** [comment, marked as answer]
I will document how to get it running locally: https://github.com/nextcloud/all-in-one/issues/878
